packages/wasi/src/index.ts

```diff
@@ -1224,7 +1224,7 @@ export default class WASIDefault {
             }
           }
           let realfd
-          if (fs.statSync(full).isDirectory()) {
+          if (!write && fs.statSync(full).isDirectory()) {
```

Comment:
Can you add a quick comment in the code saying why the `!write` is needed? Thanks!

Reply:
Is this one ok? I also added an explanation for why the `fs.statSync` call was added in the first place.